src/storage/storage_structure/node_column.cpp

```diff
@@ -0,0 +1,468 @@
+#include "storage/storage_structure/node_column.h"
+
+#include "storage/storage_structure/storage_structure.h"
+#include "storage/storage_structure/struct_node_column.h"
+#include "storage/storage_structure/var_sized_node_column.h"
+
+using namespace kuzu::catalog;
+using namespace kuzu::common;
+using namespace kuzu::transaction;
+
+namespace kuzu {
+namespace storage {
+
+void FixedSizedNodeColumnFunc::readValuesFromPage(uint8_t* frame, PageElementCursor& pageCursor,
+    ValueVector* resultVector, uint32_t posInVector, uint32_t numValuesToRead) {
+    auto numBytesPerValue = resultVector->getNumBytesPerValue();
+    memcpy(resultVector->getData() + posInVector * numBytesPerValue,
+        frame + pageCursor.elemPosInPage * numBytesPerValue, numValuesToRead * numBytesPerValue);
+}
+
+void FixedSizedNodeColumnFunc::writeValuesToPage(
+    uint8_t* frame, uint16_t posInFrame, ValueVector* vector, uint32_t posInVector) {
+    auto numBytesPerValue = vector->getNumBytesPerValue();
+    memcpy(frame + posInFrame * numBytesPerValue,
+        vector->getData() + posInVector * numBytesPerValue, numBytesPerValue);
+}
+
+void FixedSizedNodeColumnFunc::readInternalIDValuesFromPage(uint8_t* frame,
+    PageElementCursor& pageCursor, ValueVector* resultVector, uint32_t posInVector,
+    uint32_t numValuesToRead) {
+    auto resultData = (internalID_t*)resultVector->getData();
+    for (auto i = 0u; i < numValuesToRead; i++) {
+        auto posInFrame = pageCursor.elemPosInPage + i;
+        resultData[posInVector + i].offset = *(offset_t*)(frame + (posInFrame * sizeof(offset_t)));
+    }
+}
+
+void FixedSizedNodeColumnFunc::writeInternalIDValuesToPage(
+    uint8_t* frame, uint16_t posInFrame, ValueVector* vector, uint32_t posInVector) {
+    auto relID = vector->getValue<relID_t>(posInVector);
+    memcpy(frame + posInFrame * sizeof(offset_t), &relID.offset, sizeof(offset_t));
+}
+
+void NullNodeColumnFunc::readValuesFromPage(uint8_t* frame, PageElementCursor& pageCursor,
+    ValueVector* resultVector, uint32_t posInVector, uint32_t numValuesToRead) {
+    for (auto i = 0u; i < numValuesToRead; i++) {
+        bool isNull = *(frame + pageCursor.elemPosInPage + i);
+        resultVector->setNull(posInVector + i, isNull);
+    }
+}
+
+void NullNodeColumnFunc::writeValuesToPage(
```

Comment:
Aren't these writeValuesToPage functions supposed to write multiple values to the page? They all seem to be writing a single value only. Maybe rename it to writeValueToPage or writeSingleValueToPage?